File: 1431-Kids With the Greatest Number of Candies/c++/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> solve;
        int maxCandies = *max_element(candies.begin(), candies.end());
        
        for(int candy: candies) {
            if((candy + extraCandies) < maxCandies) 
                solve.push_back(false);
            else 
                solve.push_back(true);
        }
        return solve;
    }
};

// 0 ms, faster than 100.00% of C++ online submissions for Kids With the Greatest Number of Candies.
// 8.9 MB, less than 90.68% of C++ online submissions for Kids With the Greatest Number of Candies.

File: 1431-Kids With the Greatest Number of Candies/golang/solution.go
package golang

func kidsWithCandies(candies []int, extraCandies int) []bool {
	solve := make([]bool, len(candies))
	var maxCandies int = 0
	for _, candy := range candies {
		if maxCandies < candy {
			maxCandies = candy
		}
	}

	for idx, candy := range candies {
		if (candy + extraCandies) < maxCandies {
			solve[idx] = false
		} else {
			solve[idx] = true
		}
	}
	return solve
}

// 0 ms, faster than 100.00% of Go online submissions for Kids With the Greatest Number of Candies.
// 2.3 MB, less than 80.00% of Go online submissions for Kids With the Greatest Number of Candies.
